core/distributed/matrix.cpp

```diff
@@ -289,10 +296,18 @@ mpi::request Matrix<ValueType, LocalIndexType, GlobalIndexType>::communicate(
     auto recv_ptr = use_host_buffer ? host_recv_buffer_->get_values()
                                     : recv_buffer_->get_values();
     exec->synchronize();
+#ifdef GINKGO_FORCE_SPMV_BLOCKING_COMM
+    comm.all_to_all_v(use_host_buffer ? exec->get_master() : exec, send_ptr,
+                      send_sizes_.data(), send_offsets_.data(), type.get(),
+                      recv_ptr, recv_sizes_.data(), recv_offsets_.data(),
+                      type.get());
+    return {};
+#else
```

Comment:
Ah, okay, I see that you mention that is only for OpenMPI version 4.0.x ?

Reply:
Yes, but honestly, I can't tell you why this results in a deadlock with openMPI v4.0.x. It works on newer openMPI versions, and also with mvapich2, cray mpi, and intel mpi.

Reply:
I see. It probably is some bug in that version. I guess we could report it if we manage to get a small reproducible example. But given that it is an older version, I think that is not strictly necessary. We should note it somewhere in our [Known Issues page](https://github.com/ginkgo-project/ginkgo/wiki/Known-Issues) so that if someone faces this issue, they know that it is not an error from our side.

Reply:
I already put out a CMake warning, when we (hopefully correctly) detect openMPI v4.0.x. I can add it to the wiki page.